Features/add csp method

FranziPl opened the pull request:
To validate the component for the concentrating solar power, we requested data from a real power plant. They characterise their collectors in a different way, so I added a second method to calculate the thermal losses and the efficiency. I changed the examples accordingly.

jnnr commented on src/oemof/thermal/concentrating_solar_power.py:
```diff
@@ -265,16 +272,28 @@ def calc_eta_c(eta_0, c_1, c_2, iam,
-            if eta > 0:
-                eta_c[index] = eta
+    if loss_method == 'Janotte':
+        delta_t = (temp_collector_inlet + temp_collector_outlet) / 2 - temp_amb
```
delta_temp would be more consistant.

FranziPl replied:
I changed it

FranziPl commented:
Because of developing two topics at the same time, I have more changes in this PR, than the header indicates:

- I added a second method
- I added tests for the csp
- I changed the part of the code for the calc_eta_c to a better kind.

@jnnr Could you please review this PR again, so I can close it? Thank you.

jnnr commented:
Some more comments:

* in concentrating_solar_power.py:
  * The dataframe df already has a timeindex, which is internally replaced. This can lead to errors. 
  * Passing df and then telling which column means what is kind of complicated.
    I mean these 3:
    date_col: string, default: 'date'
    irradiance_col: string, default: 'E_dir_hor'
    temp_amb_col: string, default: 'temp_amb'

Suggestion: Pass irradiance (E_dir_hor or dni, depending on irradiance_method) and temp_amb as pd.Series and check that it has a valid timeindex (which is necessary for the pvlib). E_dir_hor/dni should be passed as kwargs, and depending on the method it should be checked that the right keyword argument is passed.

* parameter names
  * tz: why not call it timezone?
  * x: why not call it cleanliness? I would be ok with x however if it is an established convention to name it like this.

jnnr commented:
There is a type that should lead to a failure in line 136:

pvlib.irradiance.beam_coponent

FranziPl commented on src/oemof/thermal/concentrating_solar_power.py:
```diff
@@ -57,9 +60,7 @@ def csp_precalc(df, periods,
-    a_1: numeric
-        Parameter for the incident angle modifier.
-    a_2: numeric
+    a_1, a_2, a_3, a_4, a_5, a_6: numeric
```
I added an explanation, in which case they are needed. Apart from that i consider "Parameter for the incident angle modifier" to be sufficient

FranziPl commented on docs/concentrating_solar_power.rst:
```diff
@@ -77,11 +77,13 @@ These arguments are used in the formulas of the function:
+                                                                                  before losses because of dirtiness are considered
 
-    :math:`X`                 :py:obj:`x`                                         Cleanliness of the collector (between 0 and 1)
+    :math:`X`                 :py:obj:`cleaniness`                                Cleanliness of the collector (between 0 and 1)
```
```suggestion
    :math:`X`                 :py:obj:`cleanliness`                                Cleanliness of the collector (between 0 and 1)
```

FranziPl commented:
Thanks a lot for your input! It was very helpful.
